lib/encodings/base32.fz

```diff
@@ -0,0 +1,169 @@
+# This file is part of the Fuzion language implementation.
+#
+# The Fuzion language implementation is free software: you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as published
+# by the Free Software Foundation, version 3 of the License.
+#
+# The Fuzion language implementation is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+# License for more details.
+#
+# You should have received a copy of the GNU General Public License along with The
+# Fuzion language implementation.  If not, see <https://www.gnu.org/licenses/>.
+
+
+# -----------------------------------------------------------------------
+#
+#  Tokiwa Software GmbH, Germany
+#
+#  Source code of Fuzion standard library feature base32
+#
+# -----------------------------------------------------------------------
+
+# Base32 encoding and decoding as defined in RFC 4648
+#
+public base32 is
+
+  # allows redefinition e.g. for base32hex
+  module get_alphabet => "ABCDEFGHIJKLMNOPQRSTUVWXYZ234567".utf8.as_array
+  module encoding_name => "base32"
+
+  # NYI: compiler does currently not optimize features without arguments to treat them like fields
+  alphabet := get_alphabet
+
+  # decode a valid base32 characters to 5 bits
+  # visibility module to allow redefinition in base32hex
```

Comment:
This comment is not necessary: visibility module to allow redefinition in base32hex.
Also it might be wrong in the future if someone decides to inherit from base32 again.

It might also be a little better design if you mark `quintet_bits` and `is_valid` as abstract and then rename base32 to e.g. rfc4648 and inherit from it twice base32/base32hex